src/Common/src/System/Dynamic/Utils/ExpressionUtils.cs

```diff
@@ -100,7 +100,7 @@ public static void ValidateArgumentTypes(MethodBase method, ExpressionType nodeK
             {
                 Expression arg = arguments[i];
                 ParameterInfo pi = pis[i];
-                arg = ValidateOneArgument(method, nodeKind, arg, pi);
+                arg = ValidateOneArgument(method, nodeKind, arg, pi, methodParamName, $"{nameof(arguments)}[{i}]");
```

Comment:
nameof() is cheap, as it's just a const string in the compiled IL.  String interpolation involves a String.Format call and associated allocations/work.